src/main.rs

```diff
@@ -48,6 +50,12 @@ struct Cli {
     /// case a rendezvous server with Ping is required. This feature will be removed once https://github.com/libp2p/rust-libp2p/issues/2109 is fixed.
     #[structopt(long)]
     ping: bool,
+    /// Activates listening on websocket with the given websocket-port
+    #[structopt(long)]
+    websocket: bool,
+    /// Port used for listening on websocket
+    #[structopt(long, required_if("websocket", "true"))]
+    websocket_port: u16,
```

Comment:
If you make this an `Option`, you could get rid of the `websocket` option. `None` means disabled, `Some` means enabled.

Could also rename to `websocket` then so the user can write:

```rust
./rendezvous-server --websocket=8080
```

To be consistent, `port` could be renamed to `tcp` or `listen-tcp`?

Reply:
I went for `listen-tcp` and `listen-websocket`